contrib/babelfishpg_tsql/src/collation.c

```diff
@@ -1748,6 +1764,60 @@ pltsql_replace_non_determinstic(text *src_text, text *from_text, text *to_text,
 	return false;
 }
 
+/* 
+ * We are processesing USE DB command
+ * Communicate the same to common extension
+ * so that collation related information gets updated
+ */ 
+void
+tsql_set_db_collation(Oid database_collation_oid)
+{
+	/* Initialise collation callbacks */
+	init_and_check_collation_callbacks();
+
+	(*collation_callbacks_ptr->set_db_collation) (database_collation_oid);
+	return;
+}
+
+/* Find the collation corresponding to a specific database */
+char*
+get_collation_name_for_db(const char* dbname)
+{
+	HeapTuple	tuple;
+	Form_sysdatabases sysdb;
+	char *collation_name;
+
+	tuple = SearchSysCache1(SYSDATABASENAME, PointerGetDatum(cstring_to_text(dbname)));
+
+	if (!HeapTupleIsValid(tuple))
+		return NULL;
```

Comment:
This should never happen. Throw error instead of returning NULL

Reply:
Done